sinks/kafka/kafka.go

```diff
@@ -0,0 +1,303 @@
+		config.Producer.RequiredAcks = sarama.WaitForAll
+	case "none":
+		config.Producer.RequiredAcks = sarama.NoResponse
+	case "local":
+		config.Producer.RequiredAcks = sarama.WaitForLocal
+	default:
+		logrus.WithField("ack_requirement", ackRequirement).Warn("Unknown ack requirement, defaulting to all")
+		config.Producer.RequiredAcks = sarama.WaitForAll
+	}
+
+	switch partitioner {
+	case "random":
+		config.Producer.Partitioner = sarama.NewRandomPartitioner
+	default:
+		config.Producer.Partitioner = sarama.NewHashPartitioner
+	}
+
+	if bufferBytes != 0 {
+		config.Producer.Flush.Bytes = bufferBytes
+	}
+	if bufferMessages != 0 {
+		config.Producer.Flush.Messages = bufferMessages
+	}
+	if bufferFrequency != 0 {
+		config.Producer.Flush.Frequency = bufferFrequency
+
+	}
+
+	config.Producer.Retry.Max = retries
+
+	return config, nil
+}
+
+// newConfiguredProducer returns a configured Sarama SyncProducer
+func newConfiguredProducer(logger *logrus.Entry, brokerString string, config *sarama.Config) (sarama.AsyncProducer, error) {
+	brokerList := strings.Split(brokerString, ",")
+
+	if len(brokerList) < 1 {
+		logger.WithField("addrs", brokerString).Error("No brokers?")
+		return nil, errors.New("No brokers in broker list")
+	}
+
+	logger.WithField("addrs", brokerList).Info("Connecting to Kafka")
+	producer, err := sarama.NewAsyncProducer(brokerList, config)
+
+	if err != nil {
+		logger.Error("Error Connecting to Kafka. client error: ", err)
+	}
+
+	return producer, nil
+}
+
+// Name returns the name of this sink.
+func (k *KafkaMetricSink) Name() string {
+	return "kafka"
+}
+
+// Start performs final adjustments on the sink.
+func (k *KafkaMetricSink) Start(cl *trace.Client) error {
+	producer, err := newConfiguredProducer(k.logger, k.brokers, k.config)
+	if err != nil {
+		return err
+	}
+	k.producer = producer
+	return nil
+}
+
+// Flush sends a slice of metrics to Kafka
+func (k *KafkaMetricSink) Flush(ctx context.Context, interMetrics []samplers.InterMetric) error {
+
+	if len(interMetrics) == 0 {
+		k.logger.Info("Nothing to flush, skipping.")
+		return nil
+	}
+
+	sendMetricStart := time.Now()
+
+	successes := int64(0)
+	for _, metric := range interMetrics {
+
+		k.logger.Debug("Emitting Metric: ", metric.Name)
+		j, err := json.Marshal(metric)
+		if err != nil {
+			k.logger.Error("Error marshalling metric: ", metric.Name)
+			k.statsd.Count("kafka.marshal.error_total", 1, nil, 1.0)
+			return err
+		}
+
+		k.producer.Input() <- &sarama.ProducerMessage{
+			Topic: k.metricTopic,
+			Value: sarama.StringEncoder(j),
+		}
+		successes++
+	}
+	k.statsd.Count("kafka.metrics_success_total", successes, nil, 1.0)
+	k.statsd.TimeInMilliseconds("kafka.metrics_flush_duration_ns", float64(time.Now().Sub(sendMetricStart).Nanoseconds()), nil, 1.0)
+
+	return nil
+}
+
+// FlushEventsChecks flushes Events and Checks
+func (k *KafkaMetricSink) FlushEventsChecks(ctx context.Context, events []samplers.UDPEvent, checks []samplers.UDPServiceCheck) {
+	// TODO
```

Comment:
If we're not doing this, can you leave a note in the README?

Reply:
From the comment, it looks like that's temporary in the interface anyway(?). We should either remove it from the interface definition or split up the interface definitions so that we don't have to implement a no-op event flusher just to flush metrics.